collection/lens.rkt

```diff
@@ -39,6 +67,21 @@
   (make-lens (λ (target) (nth target index))
              (λ (target value) (set-nth target index value))))
 
+(define (mapper-lens lens)
```

Comment:
As an aside, I would perhaps *prefer* if one could just use `map` in this way, and we could define an instance for `map` where the arguments are `lens?` and `sequence?`. Unfortunately, Racket does not (currently) have multimethods, so that plan will have to be delayed for another time...